src/main/java/net/mtrop/doom/tools/decohack/data/DEHWeapon.java

```diff
@@ -334,6 +363,27 @@ public void writeObject(Writer writer, DEHWeapon weapon) throws IOException
 
 		if (ammoPerShot != weapon.ammoPerShot)
 			writer.append("Ammo per shot = ").append(String.valueOf(ammoPerShot)).append("\r\n");
+
+		// Write out MBF21 flags explicitly, 'cause it's required by the 'standard'
+		if (flags != weapon.flags)
+		{
+			writer.append("MBF21 Bits = ");
+
+			boolean more = false;
+			for(int i = 0; i < DEHWeaponFlag.VALUES.length; i++)
+			{
+				if((flags & DEHWeaponFlag.VALUES[i].getValue()) != 0)
+				{
+					if(more)
+						writer.append(" | ");
+					writer.append(DEHWeaponFlag.VALUES[i].getMnemonic());
+					more = true;
+				}
+			}
+
+			writer.append("\r\n");
+		}
```

Comment:
In light of the counter-proposed changes to the field spec, this should be changed to write just the integer value a la `DEHThing`.